modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -91,19 +93,15 @@ TensorMechanicsAction::act()
   if (isParamValid("base_name"))
     params.set<std::string>("base_name") = getParam<std::string>("base_name");
 
-  std::string short_name = "TensorMechanics";
-
   for (unsigned int i = 0; i < dim; ++i)
   {
-    std::stringstream name;
-    name << short_name;
-    name << i;
+    std::string kernel_name = "TensorMechanics" + _base_name + Moose::stringify(i);
 
     params.set<unsigned int>("component") = i;
     params.set<NonlinearVariableName>("variable") = displacements[i];
     params.set<std::vector<AuxVariableName> >("save_in") = save_in[i];
 
-    addkernel(name.str(), params);
+    addkernel(kernel_name, params);
```

Comment:
Is there a reason for having the addkernel as a separate method here?  The other actions all directly call
`_problem-> addKernel("StressDivergenceTensors", kernel_name, params)`